Support for Django 1.11 LTS

Django 1.11 LTS will be the last Django version to support Python 2. The next major release, Django 2.0, will only support Python 3.5+. So Django 1.11.x will probably be the last supported version for LTS-to-LTS support of django-compat (1.8-1.11). See Django's [release notes](https://docs.djangoproject.com/en/1.11/releases/1.11/#python-compatibility) for more.

There are Errors when running the test against Django 1.11:

```
======================================================================
FAIL: test_get_template_loaders (compat.tests.test_compat.CompatTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/walterrenner/workspace/django-compat/compat/tests/test_compat.py", line 247, in test_get_template_loaders
    self.assertEqual(len(template_loaders), 2)
AssertionError: 1 != 2
```

I will take a look in the next days.

--- a/.travis.yml
+++ b/.travis.yml
@@ -5,12 +5,6 @@ env:
     - TOX_ENV=py27-django14
     - TOX_ENV=py27-django18
     - TOX_ENV=py27-django111
-    - TOX_ENV=py34-django14
-    - TOX_ENV=py34-django18
-    - TOX_ENV=py34-django111
-    - TOX_ENV=py35-django14
-    - TOX_ENV=py35-django18
-    - TOX_ENV=py35-django111
 
 # Currently, Python 3.5 Interpreter is only available when it is specified
 # in the travis python section. So we have to work around this with this include

--- a/.travis.yml
+++ b/.travis.yml
@@ -29,6 +29,9 @@ matrix:
   - python: "3.5"
     env:
       - TOX_ENV=py35-django111
+  allow_failures:
+  - env:
+      - TOX_ENV=py35-django14
 
 install:
     - pip install tox-travis

--- a/.travis.yml
+++ b/.travis.yml
@@ -1,62 +1,49 @@
 language: python
-sudo: false
-cache: pip
 
-# Python versions supported by Django
-#       2.5 2.6 2.7   3.2 3.3 3.4 3.5 3.6
-# 1.4    X   X   X
-# 1.6        X   X     X   X
-# 1.7            X     X   X   X
-# 1.8            X     X   X   X
-# 1.9            X             X   X
-# 1.10           X             X   X
+# this list can be easily generated py running `tox -l`
+env:
+    - TOX_ENV=py27-django14
+    - TOX_ENV=py27-django18
+    - TOX_ENV=py27-django111
+    - TOX_ENV=py34-django14
+    - TOX_ENV=py34-django18
+    - TOX_ENV=py34-django111
+    - TOX_ENV=py35-django14
+    - TOX_ENV=py35-django18
+    - TOX_ENV=py35-django111
 
+# Currently, Python 3.5 Interpreter is only available when it is specified
+# in the travis python section. So we have to work around this with this include
+# GitHub issue: https://github.com/travis-ci/travis-ci/issues/4794
 matrix:
-    include:
-        - python: 2.6
-          env: DJANGO=django==1.4.* DJANGO_NOSE=django-nose==1.4.3
-        - python: 2.6
-          env: DJANGO=django==1.6.* DJANGO_NOSE=django-nose==1.4.3
-
-        - python: 2.7
-          env: DJANGO=django==1.4.* DJANGO_NOSE=django-nose==1.4.3
-        - python: 2.7
-          env: DJANGO=django==1.6.* DJANGO_NOSE=django-nose==1.4.3
-        - python: 2.7
-          env: DJANGO=django==1.7.* DJANGO_NOSE=django-nose==1.4.4
-        - python: 2.7
-          env: DJANGO=django==1.8.* DJANGO_NOSE=django-nose==1.4.4
-        - python: 2.7
-          env: DJANGO=django==1.9.* DJANGO_NOSE=django-nose==1.4.4
-        - python: 2.7
-          env: DJANGO=django==1.10.*  DJANGO_NOSE=django-nose==1.4.4
-
-        - python: 3.3
-          env: DJANGO=django==1.6.* DJANGO_NOSE=django-nose==1.4.3
-        - python: 3.3
-          env: DJANGO=django==1.7.* DJANGO_NOSE=django-nose==1.4.4
-        - python: 3.3
-          env: DJANGO=django==1.8.* DJANGO_NOSE=django-nose==1.4.4
-
-        - python: 3.4
-          env: DJANGO=django==1.7.* DJANGO_NOSE=django-nose==1.4.4
-        - python: 3.4
-          env: DJANGO=django==1.8.* DJANGO_NOSE=django-nose==1.4.4
-        - python: 3.4
-          env: DJANGO=django==1.9.* DJANGO_NOSE=django-nose==1.4.4
-        - python: 3.4
-          env: DJANGO=django==1.10.* DJANGO_NOSE=django-nose==1.4.4
-
-        - python: 3.5
-          env: DJANGO=django==1.9.* DJANGO_NOSE=django-nose==1.4.4
-        - python: 3.5
-          env: DJANGO=django==1.10.* DJANGO_NOSE=django-nose==1.4.4
+  include:
+  - python: "3.4"
+    env:
+      - TOX_ENV=py34-django14
+  - python: "3.4"
+    env:
+      - TOX_ENV=py34-django18
+  - python: "3.4"
+    env:
+      - TOX_ENV=py34-django111
+  - python: "3.5"
+    env:
+      - TOX_ENV=py35-django14
+  - python: "3.5"
+    env:
+      - TOX_ENV=py35-django18
+  - python: "3.5"
+    env:
+      - TOX_ENV=py35-django111
 
 install:
-    - pip install "pip>=7.0" wheel
-    - pip install $DJANGO
-    - pip install $DJANGO_NOSE
-    - pip install -r requirements.txt
+    - pip install tox-travis
+    - pip install coveralls
 
 script:
-    python runtests.py
+    - tox -e $TOX_ENV
+
+sudo: false
+
+after_success:
+    coveralls

--- a/.travis.yml
+++ b/.travis.yml
@@ -17,7 +17,7 @@ matrix:
       - TOX_ENV=py35-django18
   - python: "3.5"
     env:
-      - TOX_ENV=py34-django19
+      - TOX_ENV=py35-django19
   - python: "3.5"
     env:
       - TOX_ENV=py35-django110
@@ -29,7 +29,7 @@ matrix:
       - TOX_ENV=py36-django18
   - python: "3.6"
     env:
-      - TOX_ENV=py34-django19
+      - TOX_ENV=py36-django19
   - python: "3.6"
     env:
       - TOX_ENV=py36-django110

--- a/.travis.yml
+++ b/.travis.yml
@@ -2,8 +2,9 @@ language: python
 
 # this list can be easily generated py running `tox -l`
 env:
-    - TOX_ENV=py27-django14
     - TOX_ENV=py27-django18
+    - TOX_ENV=py27-django19
+    - TOX_ENV=py27-django110
     - TOX_ENV=py27-django111
 
 # Currently, Python 3.5 Interpreter is only available when it is specified
@@ -11,18 +12,30 @@ env:
 # GitHub issue: https://github.com/travis-ci/travis-ci/issues/4794
 matrix:
   include:
-  - python: "3.4"
-    env:
-      - TOX_ENV=py34-django18
-  - python: "3.4"
-    env:
-      - TOX_ENV=py34-django111
   - python: "3.5"
     env:
       - TOX_ENV=py35-django18
   - python: "3.5"
     env:
+      - TOX_ENV=py34-django19
+  - python: "3.5"
+    env:
+      - TOX_ENV=py35-django110
+  - python: "3.5"
+    env:
       - TOX_ENV=py35-django111
+  - python: "3.6"
+    env:
+      - TOX_ENV=py36-django18
+  - python: "3.6"
+    env:
+      - TOX_ENV=py34-django19
+  - python: "3.6"
+    env:
+      - TOX_ENV=py36-django110
+  - python: "3.6"
+    env:
+      - TOX_ENV=py36-django111
 
 install:
     - pip install tox-travis

--- a/.travis.yml
+++ b/.travis.yml
@@ -13,27 +13,16 @@ matrix:
   include:
   - python: "3.4"
     env:
-      - TOX_ENV=py34-django14
-  - python: "3.4"
-    env:
       - TOX_ENV=py34-django18
   - python: "3.4"
     env:
       - TOX_ENV=py34-django111
   - python: "3.5"
     env:
-      - TOX_ENV=py35-django14
-  - python: "3.5"
-    env:
       - TOX_ENV=py35-django18
   - python: "3.5"
     env:
       - TOX_ENV=py35-django111
-  allow_failures:
-  - env:
-      - TOX_ENV=py35-django14
-  - env:
-      - TOX_ENV=py34-django14
 
 install:
     - pip install tox-travis

--- a/.travis.yml
+++ b/.travis.yml
@@ -32,6 +32,8 @@ matrix:
   allow_failures:
   - env:
       - TOX_ENV=py35-django14
+  - env:
+      - TOX_ENV=py34-django14
 
 install:
     - pip install tox-travis

--- a/MANIFEST
+++ b/MANIFEST
@@ -1,5 +1,0 @@
-include LICENSE
-include README.md
-include README.rst
-include README
-include requirements.txt

--- a/MANIFEST.in
+++ b/MANIFEST.in
@@ -1,3 +1,4 @@
-include LICENSE
 include README.md
+include LICENSE
+include AUTHORS.txt
 include requirements.txt

--- a/README.md
+++ b/README.md
@@ -4,7 +4,7 @@ django-compat
 [![Build Status](https://travis-ci.org/arteria/django-compat.svg?branch=master)](https://travis-ci.org/arteria/django-compat)
 [![Stories in Ready](https://badge.waffle.io/arteria/django-compat.png?label=ready&title=Ready)](https://waffle.io/arteria/django-compat)
 
-Forward and backwards compatibility layer for Django 1.4, 1.7, 1.8, 1.9, and 1.10
+Forward and backwards compatibility layer for Django ~~1.4~~, ~~1.7~~, 1.8, 1.9, 1.10 and 1.11
 
 ~~Consider [django-compat](https://github.com/arteria/django-compat) as an experiment based on the discussion [on reddit](http://redd.it/2jrr4l). Let's see where it goes.~~
 
@@ -41,79 +41,79 @@ See a full example [here](https://github.com/arteria/django-hijack/blob/4966d886
 
 # Compatible objects
 
-|Compatible object|Specifically tested|1.4|1.7|1.8|1.9|1.10|Notes|
-|---|---|---|---|---|---|---|---|
-|`BytesIO`|:heavy_multiplication_x:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:||
-|`DjangoJSONEncoder`|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:||
-|`EmailValidator`|:heavy_multiplication_x:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:||
-|`GenericForeignKey`|:heavy_multiplication_x:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:x:|:x:||
-|`models.GenericForeignKey`|:heavy_multiplication_x:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:||
-|`HttpResponseBase`|:heavy_multiplication_x:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:||
-|`JsonResponse`|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:||
-|`LocaleRegexProvider`|:heavy_multiplication_x:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:||
-|`LocaleRegexURLResolver`|:heavy_multiplication_x:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:||
-|`NoReverseMatch`|:heavy_multiplication_x:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:||
-|`RegexURLPattern`|:heavy_multiplication_x:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:||
-|`RegexURLResolver`|:heavy_multiplication_x:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:||
-|`Resolver404`|:heavy_multiplication_x:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:||
-|`ResolverMatch`|:heavy_multiplication_x:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:||
-|`SortedDict`|:heavy_multiplication_x:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:||
-|`StringIO`|:heavy_multiplication_x:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:||
-|`URLValidator`|:heavy_multiplication_x:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:||
-|`VariableNode`|:heavy_multiplication_x:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:||
-|`View`|:heavy_multiplication_x:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:||
-|`add_to_builtins`|:heavy_multiplication_x:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:x:|:x:||
-|`admin_utils`|:heavy_multiplication_x:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:||
-|`atomic`|:heavy_multiplication_x:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:||
-|`clean_manytomany_helptext`|:heavy_multiplication_x:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:||
-|`clear_url_caches`|:heavy_multiplication_x:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:||
-|`close_connection`|:heavy_multiplication_x:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:||
-|`commit`|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:||
-|`commit_on_success`|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|`commit_on_success` replaced by `atomic` in Django >= 1.8|
-|`force_text`|:heavy_multiplication_x:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:||
-|`format_html`|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:||
-|`get_callable`|:heavy_multiplication_x:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:||
-|`get_current_site`|:heavy_multiplication_x:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:||
-|`get_ident`|:heavy_multiplication_x:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:||
-|`get_mod_func`|:heavy_multiplication_x:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:||
-|`get_model`|:heavy_multiplication_x:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:||
-|`get_model_name`|:heavy_multiplication_x:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:||
-|`get_ns_resolver`|:heavy_multiplication_x:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:||
-|`get_resolver`|:heavy_multiplication_x:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:||
-|`get_script_prefix`|:heavy_multiplication_x:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:||
-|`get_template_loaders`|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:||
-|`get_urlconf`|:heavy_multiplication_x:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:||
-|`get_user_model`|:heavy_multiplication_x:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:||
-|`get_username_field`|:heavy_multiplication_x:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:||
-|`handler404`|:heavy_multiplication_x:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:||
-|`handler500`|:heavy_multiplication_x:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:||
-|`import_module`|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:||
-|`import_string`|:heavy_multiplication_x:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:||
-|`include`|:heavy_multiplication_x:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:||
-|`is_valid_path`|:heavy_multiplication_x:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:||
-|`parse_qs`|:heavy_multiplication_x:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:||
-|`patterns`|:heavy_multiplication_x:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:x:||
-|`python_2_unicode_compatible`|:heavy_multiplication_x:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:||
-|`render_to_string`|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|The new function signature (https://docs.djangoproject.com/en/1.9/releases/1.8/#dictionary-and-context-instance-arguments-of-rendering-functions) is backported to pre-1.8.|
-|`resolve`|:heavy_multiplication_x:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:||
-|`resolve_url`|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:warning:|1.10: Reversing by dotted path has been removed|
-|`reverse`|:heavy_multiplication_x:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:||
-|`reverse_lazy`|:heavy_multiplication_x:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:||
-|`rollback`|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|Transaction savepoint (sid) is required for Django < 1.8|
-|`set_script_prefix`|:heavy_multiplication_x:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:||
-|`set_urlconf`|:heavy_multiplication_x:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:||
-|`simplejson`|:heavy_multiplication_x:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:||
-|`slugify`|:heavy_multiplication_x:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:||
-|`smart_text`|:heavy_multiplication_x:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:||
-|`unquote_plus`|:heavy_multiplication_x:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:||
-|`url`|:heavy_multiplication_x:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_multiplication_x:|Function used in `urlpatterns`|
-|`templatetags.compat.url`|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|Templatetag; import with `{% load url from compat %}`|
-|`urlencode`|:heavy_multiplication_x:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:||
-|`urlparse`|:heavy_multiplication_x:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:||
-|`urlresolvers`|:heavy_multiplication_x:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:||
-|`urlunparse`|:heavy_multiplication_x:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:||
-|`user_model_label`|:heavy_multiplication_x:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:||
-|`templatetags.compat.verbatim`|:heavy_check_mark:|:warning:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|Templatetag; import with `{% load verbatim from compat %}`. 1.4: Does not allow specific closing tags, e.g. `{% endverbatim myblock %}`, and does not preserve whitespace inside tags.|
+|Compatible object|Specifically tested|1.8|1.9|1.10|1.11|Notes|
+|---|---|---|---|---|---|---|
+|`BytesIO`|:heavy_multiplication_x:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:||
+|`DjangoJSONEncoder`|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:||
+|`EmailValidator`|:heavy_multiplication_x:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:||
+|`GenericForeignKey`|:heavy_multiplication_x:|:heavy_check_mark:|:x:|:x:|:x:||
+|`models.GenericForeignKey`|:heavy_multiplication_x:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:||
+|`HttpResponseBase`|:heavy_multiplication_x:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:||
+|`JsonResponse`|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:||
+|`LocaleRegexProvider`|:heavy_multiplication_x:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:||
+|`LocaleRegexURLResolver`|:heavy_multiplication_x:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:||
+|`NoReverseMatch`|:heavy_multiplication_x:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:||
+|`RegexURLPattern`|:heavy_multiplication_x:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:||
+|`RegexURLResolver`|:heavy_multiplication_x:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:||
+|`Resolver404`|:heavy_multiplication_x:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:||
+|`ResolverMatch`|:heavy_multiplication_x:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:||
+|`SortedDict`|:heavy_multiplication_x:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:||
+|`StringIO`|:heavy_multiplication_x:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:||
+|`URLValidator`|:heavy_multiplication_x:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:||
+|`VariableNode`|:heavy_multiplication_x:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:||
+|`View`|:heavy_multiplication_x:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:||
+|`add_to_builtins`|:heavy_multiplication_x:|:heavy_check_mark:|:x:|:x:|:x:||
+|`admin_utils`|:heavy_multiplication_x:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:||
+|`atomic`|:heavy_multiplication_x:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:||
+|`clean_manytomany_helptext`|:heavy_multiplication_x:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:||
+|`clear_url_caches`|:heavy_multiplication_x:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:||
+|`close_connection`|:heavy_multiplication_x:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:||
+|`commit`|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:||
+|`commit_on_success`|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|`:heavy_check_mark:|`commit_on_success` replaced by `atomic` in Django >= 1.8|
+|`force_text`|:heavy_multiplication_x:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:||
+|`format_html`|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:||
+|`get_callable`|:heavy_multiplication_x:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:||
+|`get_current_site`|:heavy_multiplication_x:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:||
+|`get_ident`|:heavy_multiplication_x:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:||
+|`get_mod_func`|:heavy_multiplication_x:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:||
+|`get_model`|:heavy_multiplication_x:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:||
+|`get_model_name`|:heavy_multiplication_x:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:||
+|`get_ns_resolver`|:heavy_multiplication_x:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:||
+|`get_resolver`|:heavy_multiplication_x:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:||
+|`get_script_prefix`|:heavy_multiplication_x:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:||
+|`get_template_loaders`|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:||
+|`get_urlconf`|:heavy_multiplication_x:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:||
+|`get_user_model`|:heavy_multiplication_x:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:||
+|`get_username_field`|:heavy_multiplication_x:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:||
+|`handler404`|:heavy_multiplication_x:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:||
+|`handler500`|:heavy_multiplication_x:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:||
+|`import_module`|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:||
+|`import_string`|:heavy_multiplication_x:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:||
+|`include`|:heavy_multiplication_x:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:||
+|`is_valid_path`|:heavy_multiplication_x:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:||
+|`parse_qs`|:heavy_multiplication_x:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:||
+|`patterns`|:heavy_multiplication_x:|:heavy_check_mark:|:heavy_check_mark:|:x:|:x:||
+|`python_2_unicode_compatible`|:heavy_multiplication_x:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:||
+|`render_to_string`|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|The new function signature (https://docs.djangoproject.com/en/1.9/releases/1.8/#dictionary-and-context-instance-arguments-of-rendering-functions) is backported to pre-1.8.|
+|`resolve`|:heavy_multiplication_x:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:||
+|`resolve_url`|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:warning:|:warning:|1.10: Reversing by dotted path has been removed|
+|`reverse`|:heavy_multiplication_x:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:||
+|`reverse_lazy`|:heavy_multiplication_x:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:||
+|`rollback`|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|Transaction savepoint (sid) is required for Django < 1.8|
+|`set_script_prefix`|:heavy_multiplication_x:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:||
+|`set_urlconf`|:heavy_multiplication_x:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:||
+|`simplejson`|:heavy_multiplication_x:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:||
+|`slugify`|:heavy_multiplication_x:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:||
+|`smart_text`|:heavy_multiplication_x:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:||
+|`unquote_plus`|:heavy_multiplication_x:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:||
+|`url`|:heavy_multiplication_x:|:heavy_check_mark:|:heavy_check_mark:|:heavy_multiplication_x:|:heavy_multiplication_x:|Function used in `urlpatterns`|
+|`tempat.url`|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|Templatetag; import with `{% load url from compat %}`|
+|`uravy_multiplication_x:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:||
+|`urlparse`|:heavy_multiplication_x:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:||
+|`urlresolvers`|:heavy_multiplication_x:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:||
+|`urlunparse`|:heavy_multiplication_x:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:||
+|`user_model_label`|:heavy_multiplication_x:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:||
+|`templatetags.compat.verbatim`|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|Templatetag; import with `{% load verbatim from compat %}`. 1.4: Does not allow specific closing tags, e.g. `{% endverbatim myblock %}`, and does not preserve whitespace inside tags.|
 
 # Resources and references
 
@@ -141,6 +141,9 @@ Bits and bites of the following projects were re-used to build [django-compat](h
 
 
 # Changelog
+
+### 2017/04/07
+* Update existing patches for Django 1.10
 
 ### 2016/08/02
 * Update existing patches for Django 1.10

--- a/requirements-test.txt
+++ b/requirements-test.txt
@@ -1,1 +1,0 @@
-django-nose

--- a/requirements-test.txt
+++ b/requirements-test.txt
@@ -1,0 +1,1 @@
+django-nose

--- a/requirements.txt
+++ b/requirements.txt
@@ -1,2 +1,1 @@
-six
-django-nose
+six>=1.10.0

--- a/runtests.py
+++ b/runtests.py
@@ -42,11 +42,20 @@ def setup():
         os.path.join(BASE_DIR, 'compat/tests/templates/')
     ]
 
+    _LOADERS = [
+        'django.template.loaders.filesystem.Loader',
+        'django.template.loaders.app_directories.Loader'
+    ]
+
+    TEMPLATE_LOADERS = _LOADERS
+
     TEMPLATES = [
         {
             'BACKEND': 'django.template.backends.django.DjangoTemplates',
-            'APP_DIRS': True,
             'DIRS': TEMPLATE_DIRS,
+            'OPTIONS': {
+                'loaders': _LOADERS
+            },
         },
     ]
 

--- a/setup.py
+++ b/setup.py
@@ -41,12 +41,6 @@ except (OSError, IOError):
         long_description=read('README.md')
     except (OSError, IOError):
         long_description = ""
-    
-    
-install_requires = [
-    'django>=1.4,<1.11',
-    'six>=1.10.0',
-]
 
 setup(
     name="django-compat",
@@ -58,7 +52,7 @@ setup(
     description="For- and backwards compatibility layer for Django 1.4, 1.7, 1.8, 1.9, and 1.10",
     long_description=long_description,
     license='MIT',
-    install_requires=install_requires,
+    install_requires=open('requirements.txt').read().splitlines(),
     url="https://github.com/arteria/django-compat",
     classifiers=[
         'Development Status :: 5 - Production/Stable',

--- a/tox.ini
+++ b/tox.ini
@@ -1,16 +1,13 @@
 [tox]
 envlist =
-    py27-dj{14,18,111},
-    py{33,34}-dj{18,111},
+    py{27,35,35}-django{18,19,110,111},
 
 [testenv]
 deps =
-    django14: Django>=1.4,<1.5
-    django14: django-nose==1.4.3
     django18: Django>=1.8,<1.9
-    django18: django-nose==1.4.4
+    django19: Django>=1.1,<1.10
+    django110: Django>=1.10,<1.11
     django111: Django>=1.11,<1.12
-    django111: django-nose==1.4.4
     -r{toxinidir}/requirements-test.txt
     -r{toxinidir}/requirements.txt
 commands =

--- a/tox.ini
+++ b/tox.ini
@@ -13,4 +13,5 @@ deps =
     -r{toxinidir}/requirements-test.txt
     -r{toxinidir}/requirements.txt
 commands =
+    pip freeze
     python runtests.py

--- a/tox.ini
+++ b/tox.ini
@@ -1,6 +1,6 @@
 [tox]
 envlist =
-    py{27,35,35}-django{18,19,110,111},
+    py{27,35,36}-django{18,19,110,111},
 
 [testenv]
 deps =

--- a/tox.ini
+++ b/tox.ini
@@ -1,36 +1,13 @@
-# To run the unit tests against multiple python versions you will need
-# the necessary python versions installed, and Tox.
-#   pip install tox
-#   tox
-
 [tox]
 envlist =
-    py26-django14,
-    py26-django16,
-
-    py27-django14,
-    py27-django16,
-    py27-django17,
-    py27-django18,
-
-    py32-django16,
-    py32-django17,
-    py32-django18,
-
-    py33-django16,
-    py33-django17,
-    py33-django18,
-
-    py34-django16,
-    py34-django17,
-    py34-django18,
+    py{27,34,35}-django{14,18,111}
 
 [testenv]
 deps =
-    -rrequirements.txt
-    django14: Django>=1.4,<1.5
-    django16: Django>=1.6,<1.7
-    django17: Django>=1.7,<1.8
+    django14: Django>=1.8,<1.9
     django18: Django>=1.8,<1.9
+    django111: Django>=1.11,<1.12
+    -r{toxinidir}/requirements-test.txt
+    -r{toxinidir}/requirements.txt
 commands =
     python runtests.py

--- a/tox.ini
+++ b/tox.ini
@@ -4,9 +4,12 @@ envlist =
 
 [testenv]
 deps =
-    django14: Django>=1.8,<1.9
+    django14: Django>=1.4,<1.5
+    django14: django-nose==1.4.3
     django18: Django>=1.8,<1.9
+    django18: django-nose==1.4.4
     django111: Django>=1.11,<1.12
+    django111: django-nose==1.4.4
     -r{toxinidir}/requirements-test.txt
     -r{toxinidir}/requirements.txt
 commands =

--- a/tox.ini
+++ b/tox.ini
@@ -1,6 +1,7 @@
 [tox]
 envlist =
-    py{27,34,35}-django{14,18,111}
+    py27-dj{14,18,111},
+    py{33,34}-dj{18,111},
 
 [testenv]
 deps =

